tensorboard/plugins/interactive_inference/witwidget/notebook/visualization.py

```diff
@@ -499,6 +499,27 @@ def set_compare_custom_predict_fn(self, predict_fn):
       self.set_compare_model_name('2')
     return self
 
+  def set_custom_distance_fn(self, distance_fn):
+    """Sets a custom function for distance computation.
+
+    WIT can directly use a custom function for all distance computations within
+    the tool. In this case, the provided function should accept a query example
+    proto and a list of example protos to compute the distance against and
```

Comment:
do we have the ability for the model to take provided json and not examples? i forget. if we do, we would want to do that here as well.

Reply:
I think this is resolved with your other comment, we will add json ability in a future PR.